src/repart/repart.c

```diff
@@ -7486,6 +7490,19 @@ static int context_fstab(Context *context) {
                 }
         }
 
+        if (arg_merge_fstab) {
+                _cleanup_free_ char *h = NULL;
+
+                r = read_one_line_file(path, &h);
+                if (r > 0 && strcmp(h, fstab_hdr) != 0) {
+                        r = read_full_file(path, &c, &l);
+                        if (r >= 0 && l > 0) {
+                                fprintf(f, "%s", c);
+                                (void) unlink(path);
```

Comment:
instead of unlinking just pass LINK_TMPFILE_REPLACE via the flag param of the flink_tmpfile() call below in case arg_merge_fstab is set

Reply:
Indeed. Done.